asllib/doc/Types.tex

```diff
@@ -227,12 +237,10 @@ \subsubsection{Formally}
 \end{mathpar}
 
 \begin{mathpar}
-\inferrule[well\_constrained\_empty]{
-  \constraints \eqname \emptylist
-}{
+\inferrule[pending\_constrained]{}{
```

Comment:
I think you need to fix the premises of the case above, otherwise both rules are simultaneously active.